### PR TITLE
Upgrade 0.28

### PR DESCRIPTION
**Pull request by antopalidi**
the review requires the previous PR to be merged #77 

Screenshot 2024-02-29 at 15 47 40

Screenshot 2024-02-29 at 15 47 32

Screenshot 2024-02-29 at 15 47 21

Screenshot 2024-02-29 at 16 39 23

Screenshot 2024-02-29 at 18 24 58

Screenshot 2024-02-29 at 18 22 46

Screenshot 2024-02-29 at 20 30 57

Screenshot 2024-04-17 at 12 47 35

### CODE REVIEW

File: app/views/decidim/vocdoni/admin/steps/_danger_zone.html.erb [hunk 1]
```diff
@@ -1,6 +1,9 @@
+<%# i18n-tasks-use t('decidim.vocdoni.admin.steps.danger_zone.messages.created') %>
+<%# i18n-tasks-use t('decidim.vocdoni.admin.steps.danger_zone.messages.paused') %>
+<%# i18n-tasks-use t('decidim.vocdoni.admin.steps.danger_zone.messages.vote') %>
```
Comment (microstudi) [inline app/views/decidim/vocdoni/admin/steps/_danger_zone.html.erb:3]: I see that we have lots of entries into the `i18n-tasks.yml` file. Maybe we can just add these there?

File: db/migrate/20240516163557_add_answers_count_to_decidim_vocdoni_questions.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# This migration comes from decidim_vocdoni (originally 20240516163557)
+
```
Comment (microstudi) [inline db/migrate/20240516163557_add_answers_count_to_decidim_vocdoni_questions.rb:4]:
```suggestion
```
I guess this comes from a copy

File: package.json [hunk 3]
```diff
@@ -28,22 +28,19 @@
...
-    "@decidim/eslint-config": "0.27.6",
-    "@decidim/stylelint-config": "0.27.6",
-    "eslint-config-prettier": "^8.5.0",
-    "eslint-config-standard": "^11.0.0",
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-standard": "^3.1.0",
+    "@decidim/dev": "0.28.0",
+    "@decidim/eslint-config": "0.28.0",
+    "@decidim/prettier-config": "0.28.0",
+    "@decidim/stylelint-config": "0.28.0",
```
Comment (microstudi) [inline package.json:40]: Should this point to 0.28.1?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "decidim-vocdoni",
-  "version": "1.0.0",
+  "version": "2.0",
```
Comment (microstudi) [inline package.json:3]:
```suggestion
  "version": "2.0.0",
```

File: spec/system/decidim/admin/admin_creates_wallet_spec.rb [hunk 5]
```diff
@@ -66,9 +66,43 @@
...
+        expect(page).to have_no_css("input[value='#{wallet.private_key}']")
+      end
+    end
+
+    context "when prod environment" do
+      before do
+        allow(Decidim::Vocdoni).to receive(:api_endpoint_env).and_return("prod")
+        allow(Decidim::Vocdoni).to receive(:vocdoni_reseller_name).and_return("Test reseller")
+        allow(Decidim::Vocdoni).to receive(:vocdoni_reseller_email).and_return("test_reseller@example.org")
+      end
```
Comment (microstudi) [inline spec/system/decidim/admin/admin_creates_wallet_spec.rb:80]: This looks repeated from an example above, can we just add the "it" spec to the previous context "when prod environment"?

File: spec/system/decidim/admin/admin_creates_wallet_spec.rb [hunk 6]
```diff
@@ -66,9 +66,43 @@
...
+        expect(page).to have_css("a[href='#{expected_href}']")
+      end
+    end
+
+    context "when stg environment" do
+      before do
+        allow(Decidim::Vocdoni).to receive(:api_endpoint_env).and_return("stg")
+        allow(Decidim::Vocdoni).to receive(:vocdoni_reseller_name).and_return("Test reseller")
+        allow(Decidim::Vocdoni).to receive(:vocdoni_reseller_email).and_return("test_reseller@example.org")
+      end
```
Comment (microstudi) [inline spec/system/decidim/admin/admin_creates_wallet_spec.rb:98]: Same here

File: app/packs/src/decidim/vocdoni/vocdoni_application.js
Comment (microstudi) [inline app/packs/src/decidim/vocdoni/vocdoni_application.js:1]: If this file is empty? shoudnt' we remove it along with all references to it?

File: app/cells/decidim/vocdoni/voting_step_navigation/show.erb [hunk 7]
```diff
@@ -1,29 +1,28 @@
...
-    <% end %>
-
-    <span class="button focus__next disabled-continue disabled">
-      <%= button_continue_text %>
-    </span>
-
+<%= append_javascript_pack_tag "decidim_vocdoni_voter_new_vote" %>
+<div class="form__wrapper-block flex-col-reverse md:flex-row justify-between">
+  <% if first_step? %>
+    <a></a>
```
Comment (microstudi) [inline app/cells/decidim/vocdoni/voting_step_navigation/show.erb:4]: this looks strange (I know that was there before), but do you know the reason why we put this empty link here?

**Review by microstudi — CHANGES_REQUESTED** [review]
Very heavy work Anna, congratulations!
I left some minor comments just to clarify some things and some text changes.
After we sort that out, we are ready to merge!

File: Rakefile [hunk 8]
```diff
@@ -3,6 +3,7 @@
 require "decidim/dev/common_rake"
 
 def install_module(path)
+  FileUtils.cp("babel.config.json", "#{path}/babel.config.json")
```
[the remark this answers is not in the document]
Comment (antopalidi) [reply Rakefile:6]: I checked. It's still needed, otherwise there are errors

**Pull request merged**